pkg/recipes/error.go

```diff
@@ -50,11 +53,29 @@ func NewRecipeError(code string, message string, deploymentStatus util.RecipeDep
 	return err
 }
 
-// GetRecipeErrorDetails is used to get ErrorDetails if error is of type RecipeError else returns nil.
-func GetRecipeErrorDetails(err error) *v1.ErrorDetails {
-	recipeError, _ := err.(*RecipeError)
-	if recipeError != nil {
-		return &recipeError.ErrorDetails
+// GetErrorDetails is used to get ErrorDetails from different error types.
+func GetErrorDetails(err error) *v1.ErrorDetails {
+	if err == nil {
+		return nil
+	}
+
+	switch v := err.(type) {
+	case *RecipeError:
+		return &v.ErrorDetails
+
+	case *azcore.ResponseError:
+		data := v1.ErrorResponse{}
+		c, err := io.ReadAll(v.RawResponse.Body)
+		if err != nil {
+			return &v1.ErrorDetails{
+				Code:    v1.CodeInternal,
+				Message: fmt.Sprintf("Error reading Response Error : %s", err.Error()),
+			}
+		}
+
+		_ = json.Unmarshal(c, &data)
+
+		return &data.Error
```

Comment:
Can **data.Error** be nil by any chance? But it wouldn't make the function panic I guess since we have the data defined above as ErrorResponse.